salt/states/dockerng.py

```diff
@@ -1580,6 +1580,13 @@ def running(name,
         image = ':'.join(_get_repo_tag(str(image)))
 
     if image not in __salt__['dockerng.list_tags']():
+        if __opts__['test']:
+            ret['result'] = None
+            ret['comment'] = (
+                'Image {} will be pulled and container {} will be created'
+                .format(image, name)
+            )
+            return ret
```

Comment:
I believe you shouldn't return so early, as other actions are expected from `dockerng.running` state once the image is pulled.